tests/interop/common/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{env, fmt::Display, str::FromStr};
+
+/// `Common` provides a crate with functionality that other TLS implementors
+/// might find useful if they are implementing a rust shim.
+
+/// This message is send to the server at the start of several test cases
+pub const CLIENT_GREETING: &str = "i am the client. nice to meet you server.";
+/// This short message is sent after the client greeting in the "request-response"
+/// scenarios.
+pub const SERVER_RESPONSE: &str = "i am the server. a pleasure to make your acquaintance.";
+/// The amount of data that will be downloaded by the large download test. Note 
+/// that the interop tests use a GB as 1_000^3 bytes, not 1_024^3 bytes
+pub const LARGE_DATA_DOWNLOAD_GB: u64 = 256;
+/// If a server or client doesn't support a test case, then the process should
+/// exit with this value.
+pub const UNIMPLEMENTED_RETURN_VAL: i32 = 127;
+
+pub enum PemType {
+    CaCert,
+    ServerChain,
+    ServerKey,
+    ClientChain,
+    ClientKey,
+}
+
+pub fn pem_directory() -> &'static str {
+    concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/..",
+        "/certificates/"
+    )
+}
+
+pub fn pem_file_path(file: PemType) -> &'static str {
+    match file {
+        PemType::CaCert => concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/..",
+            "/certificates/ca-cert.pem"
+        ),
+        PemType::ServerChain => concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/..",
+            "/certificates/server-chain.pem"
+        ),
+        PemType::ServerKey => concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/..",
+            "/certificates/server-key.pem"
+        ),
+        PemType::ClientChain => concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/..",
+            "/certificates/client-cert.pem"
+        ),
+        PemType::ClientKey => concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/..",
+            "/certificates/client-key.pem"
+        ),
+    }
+}
+
+/// This method is used to parse the server arguments from the command line argv.
+///
+/// It will the return the [InteropTest] that is being run, as well as the expected
+/// port for the server to run on.
+pub fn parse_server_arguments() -> (InteropTest, u16) {
```

Comment:
Hmm, I'm a little wary of doing that, because while rust makes it super easy to have elegant argument parsing, that is not the case for other languages (Go, Java) so we'd have to implement our own argument parsing in those languages. That argument parsing is likely just going to be `test = argv[2]` and `port = argv[4]`, at which point there didn't seem to be much advantage in specifying additional arguments.